homeassistant/util/icon.py

```diff
@@ -0,0 +1,17 @@
+"""Icon util methods."""
+from typing import Optional
+
+
+def icon_for_battery_level(battery_level: Optional[int]=None,
+                           charging: bool=False) -> str:
+    """Return a battery icon valid identifier."""
+    icon = 'mdi:battery'
+    if battery_level is None:
+        return icon + '-unknown'
+    if charging:
+        icon += '-charging'
+    if 20 <= battery_level < 100:
+        icon += '-{}'.format(int(battery_level / 20) * 20)
```

Comment:
This does not take into account the missing `battery-charging-30`, `battery-charging-50` and `battery-charging-70` icons.
https://materialdesignicons.com/